service/storage_proxy.cc

```diff
@@ -3466,7 +3466,7 @@ storage_proxy::mutate_atomically_result(std::vector<mutation> mutations, db::con
                             auto local_addr = utils::fb_utilities::get_broadcast_address();
                             auto& topology = _tmptr->get_topology();
                             auto local_dc = topology.get_datacenter();
-                            auto& local_endpoints = topology.get_datacenter_racks().at(local_dc);
+                            auto local_endpoints = topology.get_datacenter_rack_endpoints(local_dc);
```

Comment:
Why not capture by reference? Do we not hold a topology owning pointer here?

Reply:
hmm, true that we can at this time.
a following patch (not in this series) calculates those too and will return a `std::unordered_map<sstring, std::unordered_set<inet_address>>` by value, unless we keep it precalculated, or calculated once on demand and caching the result.

Reply:
You can use `auto&&` for these cases. If a reference is returned it will be a reference. If a temporary is returned then `auto&&` will extend the lifetime of the temporary until the end of the scope.